### DISCUSSION TITLE
Is it possible to embed a element inside a link(standard.link)

### DISCUSSION THREAD

**Discussion by reshin-pulapaddy** [clientIO/joint#1980] [Q&A] [answered]
Hi @kumilingus 
Is it possible to embed an element inside a link(standard.link)?
**requirement**
when I change the position of a link I need to automatically change the element position also.
It works when I embed a link inside an element

`    this.paper.on("link:pointerdown", (elementView, evt) => {
      console.log("link ponter dowm");
      console.log(elementView);
      // Select an LINK if CTRL/Meta key is pressed while the link is clicked.
      var rect = new joint.shapes.standard.Rectangle();
        rect.position(200, 400);
        rect.resize(100, 40);
        rect.attr({
            body: {
                fill: 'blue'
            },
            label: {
                text: 'Hello',
                fill: 'white'
            }
        });
        rect.addTo(this.graph)
        console.log(rect.id);
        elementView.model.embed(rect)
        console.log("el",elementView);
        this.selection.collection.reset([elementView.model]);`

**Comment by kumilingus** [comment, marked as answer]
It is possible to embed an element into a link. However, the element will not move with the link.

Only link labels move with the link. The position of each label on the link is clearly defined using relative or absolute `distance` and `offset. In the case of embedding, it is not clear how to update the elements after the link changes shape.

Therefore, you must either use link labels (and implement a custom selection), or listen for changes and update the element position yourself.

```js
const link = new joint.shapes.standard.Link({
    source: { x: 50, y: 500 },
    target: { x: 500, y: 500 },
});

const element = new joint.shapes.standard.Rectangle({
    size: { width: 100, height: 100 },
    position: {  x: 100, y: 100 }
});

graph.addCells([link, element]);

graph.on('change:source change:target change:vertices', function() {
    const { x, y } = link.getPolyline().pointAt(0.5);
    const { width, height } = element.size();
    element.position(x - width / 2, y - height / 2);
});
```
Here's a working demo: https://jsfiddle.net/kumilingus/q6zf7Lcy/

**Reply by reshin-pulapaddy** [reply]
issue solved. 
Thank you @kumilingus
